improver/psychrometric_calculations/precip_phase_probability.py

```diff
@@ -0,0 +1,182 @@
+
+    For snow; the 80th percentile is taken from a neighbourhood around
+    each point and is compared with the orography. Where the orography is
+    higher, the returned probability-of-snow is 1, else 0.
+    For rain, the above method is modified to get the 20th percentile
+    and where the orography is lower than the percentile value, the returned
+    probability-of-rain is 1, else 0.
+    """
+    def __init__(self, radius=10000.):
+        """
+        Initialise plugin
+        Args:
+            radius (float):
+                Neighbourhood radius from which 80th percentile is found (m)
+        """
+        self.percentile_plugin = GeneratePercentilesFromANeighbourhood
+        self._nbhood_shape = 'circular'
+        self.radius = radius
+
+    def _extract_input_cubes(self, cubes):
+        """
+        Separates the input list into the required cubes for this plugin,
+        detects whether snow or rain are required from the input phase-level
+        cube name and appropriately initialises the percentile_plugin, sets
+        the appropriate comparator operator for comparing with orography and
+        the unique part of the output cube name.
+
+        Converts units of falling_level_cube to that of orography_cube if
+        necessary. Sets flag for snow or rain depending on name of
+        falling_level_cube.
+
+        Args:
+            cubes (iris.cube.CubeList or list):
+                Contains cubes of the altitude of the phase-change level (this
+                can be snow->sleet, or sleet->rain) and the altitude of the
+                orography. The name of the phase-change level cube must be
+                either "altitude_of_snow_falling_level" or
+                "altitude_of_rain_falling_level". The name of the orography
+                cube must be "surface_altitude".
+
+        Raises:
+            ValueError: If cubes with the expected names cannot be extracted.
+            ValueError: If cubes does not have the expected length of 2.
+            ValueError: If the extracted cubes do not have matching spatial
+                        coordinates.
+
+        """
+        if isinstance(cubes, list):
+            cubes = iris.cube.CubeList(cubes)
+        if len(cubes) != 2:
+            raise ValueError(f'Expected 2 cubes, found {len(cubes)}')
+
+        if not spatial_coords_match(cubes[0], cubes[1]):
+            raise ValueError('Spatial coords mismatch between '
+                             f'{cubes[0]} and '
+                             f'{cubes[1]}')
+
+        extracted_cube = cubes.extract('altitude_of_snow_falling_level')
+        if extracted_cube:
+            self.falling_level_cube, = extracted_cube
+            self.param = 'snow'
+            self.comparator = operator.gt
+            self.get_discriminating_percentile = self.percentile_plugin(
+                self._nbhood_shape, self.radius, percentiles=[80.]).process
```

Comment:
Just to say that you don't actually need the `.process` here, due to the behaviour of the BasePlugin, so this could be: 
```
            self.get_discriminating_percentile = self.percentile_plugin(
                self._nbhood_shape, self.radius, percentiles=[80.])
```

Reply:
I didn't know you could do this. Something learned. Code changed.